WordPress/Classes/Services/MediaImageService.swift

```diff
@@ -276,14 +386,37 @@ extension MediaImageService {
     }
 }
 
-// MARK: - Helpers (RemoteURL)
+// MARK: - SafeMedia
+
+/// A thread-safe media wrapper for use by `MediaImageService`.
+private final class SafeMedia {
```

Comment:
Using a class is more efficient in this case because the `SafeMedia` instance gets passed between multiple methods. If it was a struct, it would require a lot of copying and would introduce ARC overhead on a per-property level for this struct. So I specifically decided to go with a class.